src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/AdherenceRecord.java

```diff
@@ -148,4 +154,16 @@ public String getAssessmentGuid() {
     public void setAssessmentGuid(String assessmentGuid) {
         this.assessmentGuid = assessmentGuid;
     }
+    public Set<String> getUploadIds() {
+        return uploadIds;
+    }
+    public void setUploadIds(Set<String> uploadIds) {
+        this.uploadIds = uploadIds;
+    }
+    public void addUploadId(String uploadId) {
+        if (uploadIds == null) {
+            uploadIds = new HashSet<>();
+        }
+        uploadIds.add(uploadId);
+    }
```

Comment:
If it's easy to do, we should create upload IDs as a separate table so that we can query on these easily. (There's a way to do this in Hibernate, so you don't need to create a new class, but I forget the details off the top of my head.)

If that's a big amount of work, we could serialize this as a comma-delimited list, and put a SQL index on it, which is a bit hacky, but we should still be able to query on it.

Reply:
See for example, the annotations on HibernateSharedModuleMetadata.getTags()

Reply:
Thank you for the example. It's updated and working. The compound ID for adherence records is 5 fields, so the new table is unfortunately quite large for what it is.

Another option would be to add a single generated unique ID to the adherence record that's not exposed through the API. And just use it for internal joins. Let me know if that sounds better to you and I could give it a shot in the morning.